### DISCUSSION TITLE
Is there a way to keep leading spaces from being stripped by Text component?

### DISCUSSION THREAD

**Discussion by njsdev** [diegomura/react-pdf#1695] [Q&A] [answered]
Is your feature request related to a problem? Please describe.
I have an application, where leading spaces are needed in front of text. The text component seems to strip out all leading spaces.
An example is in monospace font, I may have 2 lines of digits, the digits need to remain in place like a column. If the first digit of the first line is 11, and the second digit of the second line is 2, I would place a space in front of 2 to keep it in line with the 11.

Describe the solution you'd like
I'd like to know if you know a workaround where I can use leading spaces within the text component without it being removed.

Describe alternatives you've considered
I've tried using a < pre > tag. I'm not sure what else to try.

Additional context
I'm looking for a way to manipulate the string or the component to allow the leading space.

**Comment by jeetiss** [comment, marked as answer]
Hello @njsdev
react-pdf doesn't have pre tag or any analog of it, as a workaround you can set non-breaking space at the start of the line. here the [example](https://react-pdf.org/repl?code=3187b0760ce02e00408a057025803c450298c0bc300500943807cf805030c00f0022230080b6198531165d400a02180e659a004f003619b006f61622003a0046200099080beed3a72a005430a3601188d500f4bbf46cdd5cdb0901c800e82000cce020b3804c9eeead3372d3474f56d1c5ddd9c9d5c3c01997dfc4303a84d78052d4ce8199959d80801b8c8c9412161a4302070600194a14430aa002c3030a16580009c307930f02439145400b860fb34001c78949490c0f986620158006839545728a090a0c587473956c955f30a0094bb80a0b86800c5653ac09431daf0a91151d0b04d898e800) 

i hope it helps

**Reply by njsdev** [reply]
@jeetiss This is exactly what I was looking for. Thank you!
